### ISSUE TITLE
KS0 upgrade problem

### ISSUE THREAD

**Issue by vruzayev** [incidrthreat/iceriver_oc#6]
Hello. Please help me. 
I have 5 KS0 devices. Decided to upgrade. Started with first one. I downloaded 160 firmware, open miner GUI and started upgrade. It shows upgraded successfully . Asked to reboot
After rebooting miner booted properly and start mining. I waited few minutes until pool recognized it and continue with other miners. 
After upgrades May be 40 minutes later I found my hashrate didn't increase, still about 100 GH/s. I decided to reboot them by disconnecting power cord. After this all my ASICS weren't able to boot. I can ping them. I installed Eclipse and able to see file system of each ASIC, but don't know how I can fix them. 
Please, help me to restore normal work of my devices

With best regards
Vitaly

**Comment by JustLook12** [comment]
hi there i am not off the team but : 
1. you need to do a factory reset. 
2. after factory reset you need first to upgrade with official Iceriver firmware
3. Then you can apply update via webgui to get the oc . 

When you upgrade without the official firmware of Iceriver machine when reboot does not boot. (my expirience)

**Comment by incidrthreat** [comment]
One way to fix this is to copy original iceriver firmware from their site: https://www.iceriver.io/firmware-download/

Then using Eclipse IDE, drag and drop the `update.bgz` file into the `/var/volatile/tmp/bg/uploads` folder and reboot

If that does not work, factory reset and try a lower OC update.

**Comment by vruzayev** [comment]
Thank you for response. Wil factory reset return original firmware? Right now I'm ok even with original one, because I'm not able to mine

**Comment by JustLook12** [comment]
If you do factory reset in accordance to manual (so through the power buttom) it will restore to the firmware you machine was shipped with. So yes. Then you can upgrade with latest ICERiver firmware. I had same as you had and restored in that manner

**Comment by vruzayev** [comment]
Thank you, everyone for responses and help

**Comment by JustLook12** [comment]
> Thank you, everyone for responses and help

so problem solved ?

**Comment by vruzayev** [comment]
Solved, partially.
I reset all my ASICs and return them to normal operation. But when I try to upgrade firmware , even from Iceriver website, ASICs become inaccessible. Green led on front of device is solid, unable to connect. after reset I come back to factory settings. 
I don't know, may be I'm doing something wrong, But I think it is easy. Just choose new firmware file and click on Upgrade button. But after not able to access to device. and try to reboot many times. nothing help. Only reset.

**Comment by incidrthreat** [comment]
> Solved, partially. I reset all my ASICs and return them to normal operation. But when I try to upgrade firmware , even from Iceriver website, ASICs become inaccessible. Green led on front of device is solid, unable to connect. after reset I come back to factory settings. I don't know, may be I'm doing something wrong, But I think it is easy. Just choose new firmware file and click on Upgrade button. But after not able to access to device. and try to reboot many times. nothing help. Only reset.

Have you attempted to reconnect to it using the Eclipse IDE?

**Comment by JustLook12** [comment]
> Solved, partially. I reset all my ASICs and return them to normal operation. But when I try to upgrade firmware , even from Iceriver website, ASICs become inaccessible. Green led on front of device is solid, unable to connect. after reset I come back to factory settings. I don't know, may be I'm doing something wrong, But I think it is easy. Just choose new firmware file and click on Upgrade button. But after not able to access to device. and try to reboot many times. nothing help. Only reset.

That is strange, i only had what you mentioned (e.g. miner accessible via Eclipse IDE, but webgui not accessibel) if after a factory reset, when i tried to apply the OC (update.bgz) directly, without first upgrading to latest Iceriver firmware. From what you mention you cannot even upgrade machines with Iceriver firmware from a stock firmware. DId you managed upgrading firmware in the past from Iceriver?

**Comment by vruzayev** [comment]
> That is strange, i only had what you mentioned (e.g. miner accessible via Eclipse IDE, but webgui not accessibel) if after a factory reset, when i tried to apply the OC (update.bgz) directly, without first upgrading to latest Iceriver firmware. From what you mention you cannot even upgrade machines with Iceriver firmware from a stock firmware. DId you managed upgrading firmware in the past from Iceriver?

I tried only upgrade through web interface from Firmware upgrade link, and all time my KS0 stop working. No matter what kind of firmware I put in. It happened with original and with custom from this github. Yesterday I tried all of them starting from 120 to 160. But I still have access by Eclipse. And see file system. I never tried upgrade by Eclipse.

**Comment by vruzayev** [comment]
Unable to upgrade through Eclipse. Procedure shows need to upgrade up to latest official revision, and then replace miner.bgz
I'm not able to upgrade even to official revision. After upgrade my ASICs become not bootable.

**Comment by JustLook12** [comment]
Did you try as incidrthreat mention placed with eclipse ide the official firmware in folder and reboot through webgui?

**Comment by vruzayev** [comment]
If I upgrade to official new revision, and restart through the GUI miner booted. I tried to replace miner.bgz and reboot through GUI ASIC not booted.
What will happen if I will copy update.bgz to the update folder and reboot device? will it updated?

**Comment by vruzayev** [comment]
Tried. nothing happened.

**Comment by JustLook12** [comment]
1. So updates through gui with official iceriver it works and boots. (Check)
2. If you then take the update.bgz (oc file) (so first step 1) and reboot does it start ? 
So let me explain. If you do a factory reset your firmware becomes version 0. Miner boots. 
You need to upgrade then with iceriver firmware. Then miner has rev 1. The oc mod work with firmware rev1. So when you have updated to latest then and only then can you go for ocmodded firmware either by ide or by update160.bgz

**Comment by vruzayev** [comment]
Just updated to new official through WEB GUI and rebooted from GUI. It booted. Eclipse shows 2 files wit different names(miner and ks0bg. upgraded to 160 through GUI and rebooted through GUI. It booted. Eclipse shows same. But need 5-10 minutes to confirm new OC revision works

**Comment by vruzayev** [comment]
Quick update. I did one ASIC
no one of the revisions 120, 140, 150 and 160 not works on my ASIC. It booted after update, but hashrate stay around 100 G.
I tried very first available OC rev update.dgz without number. My ASIC on GUI reported 160, and pool reported 140-150G.
I didn't try to reboot ASIC by disconnecting power. 
Will try to make same upgrade on couple other my ASICs before go to bed.

**Comment by vruzayev** [comment]
![Capture](https://github.com/incidrthreat/iceriver_oc/assets/146872212/a909d3df-a4b7-4dce-8115-48aee2485800)
4 ASICs reported 160 G one 120. Pool reported approximately same

**Comment by JustLook12** [comment]
So problem solved.

**Comment by tomekmak** [comment]
Is there any kind of tutorial how to use Eclipse IDE to upgrade firmware ?

**Comment by JustLook12** [comment]
See the readme.md. note only 160gh/s is available to do via eclipse i think. But steps are simular. 
1. Ensure your firmware is recent official one of iceriver
2. Install eclips
3. Open eclipse (see readme above the ssh parr) to get tcf agent and be able to browse  ks0.
4. Browse to designated folder (see readme) 
5. Delete miner.bgz 
6. Upload miner.bgz
7. Power off 
8. Power on
9. Done wait should be mining in 7 minutes. Note upgrade charger.
This out of head procedure is in readme.md

**Comment by vruzayev** [comment]
> See the readme.md. note only 160gh/s is available to do via eclipse i think. But steps are simular.
> 
> 1. Ensure your firmware is recent official one of iceriver
> 2. Install eclips
> 3. Open eclipse (see readme above the ssh parr) to get tcf agent and be able to browse  ks0.
> 4. Browse to designated folder (see readme)
> 5. Delete miner.bgz
> 6. Upload miner.bgz
> 7. Power off
> 8. Power on
> 9. Done wait should be mining in 7 minutes. Note upgrade charger.
>    This out of head procedure is in readme.md

I tried to upgrade by Eclipse. Unsuccessful. But Was able to upgrade very first revision. I will leave it like that. If after power reboot ASICs will not start will upgrade them again
I think we can close the case. I will open new one if will have some problem in future

**Comment by speedyfoxai** [comment]
I have the exact same problem (no matter how I flash always ~100gh), except the original update file without numbers did not work for me... 

No matter what I flash, it always does ~100gh... I have always been able to access the GUI and I have done the full reset many times. I flash with updated ICERIVER FW, reboot, Flash update firmware, nothing.

I have seen folks respond on YT having same issue as I am too...

**Comment by JustLook12** [comment]
Then it seems that you guys are unlucky and silicon lottery does not bring any added gains in terms of oc. Sorry guys...

**Comment by JustLook12** [comment]
@mdkrush, then seems u are unlucky. The original file without numbers is meant to be done via eclips. While other files was through webgui. 
Trust your powersupply is of higher then 120w.

**Comment by vruzayev** [comment]
> @mdkrush, then seems u are unlucky. The original file without numbers is meant to be done via eclips. While other files was through webgui. Trust your powersupply is of higher then 120w.

Mine all work perfect already 4 days. 4 of them have 160 GH one 130 GH. Total pool reported 750-790 GH. I purchased 3 PS 12V 400 Amps (480 W) and use them for 2 ASICs each.
https://www.amazon.ca/dp/B071DQPRR5?ref=ppx_yo2ov_dt_b_product_details&th=1 
I think can use one for 4. I assembled RIG on regular open frame for 12 GPUs. Mount 4 ASICs and also there have space for 5 ports Ethernet hub small power bar and power supplies. Looks nice and accurate
Thank you for your help
